docs/reference/remap.cue

```diff
@@ -52,17 +51,589 @@ remap: {
+				(`$code = .status_code`), and more.
+
+				### Nested values
+
+				The dot syntax can represent nested fields, for example `.transaction.id` or
+				`.geo.latitude`. If you assign values to nested fields whose parent fields don't
+				exist, the parent fields are created. Take this expression as an example:
+
+				```
+				.user.info.hobbies = ["cooking", "Pogs"]
+				```
+
+				If the `user` field doesn't exist, it is created; if `.user` exists but `.user.info`
+				doesn't, `.user.info` is created; and so on.
+
+				### Path coalescing
+
+				Path *coalescing* in TRL enables you to express "or" logic inside of paths. This
+				expression sets `user.first_name` to `"Feldman"` *if* that field exists; if not, the
+				`user.last_name` field is set to `"Feldman"` instead.
+
+				```
+				.user.(first_name | last_name) = "Feldman"
+				```
+
+				### Quoted paths
+
+				In the examples above, all paths have used literals à la `.foo.bar`. But path
+				segments can also be quoted, as in this example:
+
+				```
+				user.preferences."favorite color" = "chartreuse"
+				```
+
+				Quoted paths are particularly useful when keys need to contain whitespace.
+
+				### Indexing
+
+				Values inside TRL arrays can be accessed via index (beginning with 0). For the array
+				`$primary = ["magenta", "yellow", "cyan"]`, `$primary[0]` would yield `"magenta"`.
+
+				You can also assign values to arrays by index:
+
+				```
+				$stooges = ["Larry", "Moe"]
+				$stooges[2] = "Curly"
+				```
+
+				You can even assign values to arbitrary indices in arrays; any indices that need to
+				be created are back-filled as `null`. For example, if the `hobbies` field doesn't
+				exist, the expression `.hobbies[2] = "Pogs"` sets `hobbies` to `[null, null,
+				"Pogs"]`.
+
+				### Combined
+
+				All of the path methods above can be combined in any way. Here's an example of a
+				complex path:
+
+				```
+				.transaction.(metadata | info).orders[0] = "a1b2c3d4e5f6"
+				```
+
+				This sets the first element of `.transaction.metadata.orders` to `"a1b2c3d4e5f6"`
+				*if* `.transaction.metadata` exist; if not, it sets `.transaction.info.orders` to
+				that value.
+				"""
+			examples: [
+				".",
+				".status_code",
+				#".message.event."time of occurrence""#,
+				".transaction.id",
+				".user.hobbies[0].description",
+				".event.(time | timestamp).format",
+				#""#,
+			]
+		}
+
+		"Expressions": {
+			href: "expressions"
+
+			description: """
+				*All* expressions in TRL resolve to a value. Expressions come in four kinds, listed
+				below, each of which resolves in a different way:
+
+				Expression type | Resolves to
+				:---------------|:-----------
+				Assignment | The assigned value
+				Control flow statements | The value returned by the chosen expression
+				Boolean expressions | The returned Boolean value
+				Blocks | The value returned by the last expression in the block
+				"""
```

Comment:
I would probably describe this a little differently.

I'd say "the smallest unit of code in TRL is an expression. Every expression resolves to a concrete value. The simplest expression is a literal expression, e.g. a numeric literal `5` resolves to the integer `5`, a string literal `"foo"` resolves to the string value `"foo"`, etc.

The list above also misses a few key expression types, namely path/variable queries, function calls, groups (e.g. `(5 + 2) * 3`), arithmetic expressions, literals, and container types (array and map).

It's probably also worth mentioning expressions are either terminated with a newline or a semicolon. There are exceptions to this, since certain expressions allow newlines at certain points, e.g. the if-statement expression:

```rust
if foo {
  .bar
}
```

Can be split over multiple lines even though it resolves to a single expression (the same applies to blocks).

Similarly, array and maps are allowed to be split over multiple lines:

```rust
.foo = [
	"bar",
	"baz",
]

.$foo = {
	"bar": 1,
	"baz": true,
}
```